jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java

```diff
@@ -73,41 +72,66 @@ private TextMapCodec(Builder builder) {
 
   static JaegerSpanContext contextFromString(String value)
       throws MalformedTracerStateStringException, EmptyTracerStateStringException {
-    if (value == null || value.equals("")) {
+    if (value == null || value.isEmpty()) {
       throw new EmptyTracerStateStringException();
     }
 
-    String[] parts = value.split(":");
-    if (parts.length != 4) {
+    final int traceIdEnd = value.indexOf(':');
+    if (traceIdEnd == -1) {
       throw new MalformedTracerStateStringException(value);
     }
 
-    String traceId = parts[0];
-    if (traceId.length() > 32 || traceId.length() < 1) {
-      throw new TraceIdOutOfBoundException("Trace id [" + traceId + "] length is not withing 1 and 32");
+    if (traceIdEnd > 32 || traceIdEnd == 0) {
+      throw new TraceIdOutOfBoundException(
+              "Trace id [" + value.substring(0, traceIdEnd) + "] length is not within 1 and 32");
     }
 
-    // TODO(isaachier): When we drop Java 1.6 support, use Long.parseUnsignedLong instead of using BigInteger.
```

Comment:
can we use Long.parseUnsignedLong now that the build is on 1.8?

Reply:
Yes, but in Java 8 it would require creating temporary String objects.
In Java 9 there's a variant of [parseUnsignedLong](https://github.com/openjdk/jdk/blob/b29fbad940808c59f30e60222a9ca7a23c8e54b9/src/java.base/share/classes/java/lang/Long.java#L996) with the following signature:
`parseUnsignedLong(CharSequence s, int beginIndex, int endIndex, int radix)`
which allows to avoid creating temporary strings.

Reply:
please restore the TODO commend with your explanation

Reply:
@yurishkuro added a comment.